### PR TITLE
Moving sub-classes to separate files 

### LINKED ISSUE #4571: Move sub-classes to their own files
**Controls that contain sub-classes:**
- [x]  ComponentEditorForm (#4572)
- [x]  DataGridViewComboBoxEditingControl (#4572)
- [x]  DocComment (#4572)
- [x]  DomainUpDown (#4573)
- [x]  DropDownButton (#4573)
- [x]  Form (#4573)
- [x]  GridErrorDlg (#4575)
- [x]  HotCommands (#4575)
- [x]  Label (#4575)
- [x]  MDIClient (#4576)
- [x]  MDIWindowDialog (#4576)
- [x]  PropertyGrid (#4602)
- [x]  PropertyGridView (#4603)
- [x]  RichTextBox (#4576)
- [x]  ScrollableControl (#4576)
- [x]  TabPage (#4694)
- [x]  ToolStrip (#4577)
- [x]  ToolStripContainer (#4577)
- [x]  ToolStripDropDown (#4577)
- [x]  ToolStripDropDownMenu (#4578)
- [x]  ToolStripPanel (#4578)
- [x]  UpDownButtons (#4579)
- [x]  WebBrowser (#4579)
- [x]  WebBrowserBase (#4579)

**Components that contain sub-classes:**
- [x]  ColumnHeader (#4605)
- [x]  ErrorProvider (#4605)
- [x]  FileDialog_Vista (#4606)
- [x]  NotifyIcon (#4612)
- [x]  OpenFileDialog (#4606)
- [x]  SaveFileDialog (#4606)
- [x]  ToolStripSplitButton  (#4612)
- [x]  ToolStripStatusLabel  (#4612)
- [x]  ToolTip (#4694)
- [x]  ToolStripLabel (#4613)
- [x]  ToolStripPanelRow (#4613)
- [x]  ToolStripProgressBar (#4614)
- [x]  ToolStripComboBox (#4614)
- [x]  ToolStripDropDownButton (#4613)
- [x]  ToolStripDropDownItem  (#4612)

**Lists and collections that contain sub-classes:**
- [x]  DataGridViewCellLinkedList (#4615)
- [x]  DataGridViewColumnCollection (#4615)
- [x]  DataGridViewIntLinkedList (#4615)
- [x]  DataGridViewRowCollection (#4617)
- [x]  ImageList.ImageCollection (#4615)
- [x]  ListBox.IntegerCollection (#4617)
- [x]  ListBox.SelectedIndexCollection (#4617)

**Classes that contain sub-classes:**
- [x]  CategoryGridEntry (#4618)
- [x]  DataFormats (#4618)
- [x]  DataGridViewComboBoxCell (#4618)
- [x]  DataGridViewHeaderCell (#4618)
- [x]  DataGridViewRow (#5136)
- [x]  DataGridViewTopLeftHeaderCell (#4695)
- [x]  FlatButtonAppearance (#4695)
- [x]  FlowLayout (#4698)
- [x]  FontCache (#4695)
- [x]  GridEntry (#4700)
- [x]  HtmlDocument (#4702)
- [x]  HtmlElement (#4702)
- [x]  HtmlWindow (#4702)
- [x]  KeyboardToolTipStateMachine (#4703)
- [x]  ListViewItemConverter (#4703)
- [x]  ListViewSubItem (#4703)
- [x]  LayoutUtils (#4703)
- [x]  MergePropertyDescriptor (#4705)
- [x]  MultiSelectRootGridEntry (#4705)
- [x]  PrintControllerWithStatusDialog (#4705)
- [x]  PropertyDescriptorGridEntry (#4706)
- [x]  PropertyStore (#4706)
- [x]  Screen (#5136)
- [x]  SendKeys (#5136)
- [x]  TableLayout (#5136)
- [x]  ToolStripItemInternalLayout (#5136)
- [x]  ToolStripSettings (#5136)
- [x]  TreeNode (#5136)

### PR DESCRIPTION
**Pull request by lisoenot**
Fixes #4571

## Proposed changes

- Moved sub-classes of "Screen", "SendKeys", "TableLayout", "ToolStripItemInternalLayout", "ToolStripSettings", "TreeNode" and "DataGridViewRow" classes to their own files

## Customer Impact

- No

## Regression? 

- No

## Risk

- Minimal

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5136)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/winforms?pullRequest=5136) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/winforms?pullRequest=5136) before we can accept your contribution.<br/><br/>:x: lisoenot [sign now](https://cla.dotnetfoundation.org/dotnet/winforms?pullRequest=5136)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/winforms?pullRequest=5136) it.</sub>

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.SpanComparer.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            public abstract int GetSpan(LayoutInfo layoutInfo);
+
+        public int Compare(object x, object y)
+        {
+            LayoutInfo xInfo = (LayoutInfo)x;
+            LayoutInfo yInfo = (LayoutInfo)y;
+            return GetSpan(xInfo) - GetSpan(yInfo);
+        }
+    }
+}
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.SpanComparer.cs:24]: Incorrect indentation

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.RowSpanComparer.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            return layoutInfo.RowSpan;
+        }
+
+        public static RowSpanComparer GetInstance
+        {
+            get { return instance; }
+        }
+    }
+}
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.RowSpanComparer.cs:23]: Incorrect indentation

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.ColumnSpanComparer.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            return layoutInfo.ColumnSpan;
+        }
+
+        public static ColumnSpanComparer GetInstance
+        {
+            get { return instance; }
+        }
+    }
+}
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.ColumnSpanComparer.cs:23]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.PostAssignedPositionComparer.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+#nullable disable
+
+using System.Collections;
+
+namespace System.Windows.Forms.Layout
+{
+    internal partial class TableLayout
+    {
+        private class PostAssignedPositionComparer: IComparer
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.PostAssignedPositionComparer.cs:13]: Add a missed space
```suggestion
        private class PostAssignedPositionComparer : IComparer
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.PreAssignedPositionComparer.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace System.Windows.Forms.Layout
+{
+    internal partial class TableLayout
+    {
+        private class PreAssignedPositionComparer: IComparer
+        {
+            private static readonly PreAssignedPositionComparer instance = new PreAssignedPositionComparer();
+
+        public static PreAssignedPositionComparer GetInstance
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.PreAssignedPositionComparer.cs:17]: Incorrect indentation

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.PreAssignedPositionComparer.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+#nullable disable
+
+using System.Collections;
+
+namespace System.Windows.Forms.Layout
+{
+    internal partial class TableLayout
+    {
+        private class PreAssignedPositionComparer: IComparer
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.PreAssignedPositionComparer.cs:13]: ditto
```suggestion
        private class PreAssignedPositionComparer : IComparer
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.SpanComparer.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.SpanComparer.cs:5]: Check if we need this here and in the rest files

**Review by vladimir-krestov — CHANGES_REQUESTED** [review]
Needs refactoring

**Review by RussKie — COMMENTED** [review]
👍

**Comment by RussKie** [comment]
@lisoenot if you think it is ready to go, please make it ready for review.

**Review by vladimir-krestov — APPROVED** [review]

**Review by RussKie — APPROVED** [review]

**Pull request merged**